src/components/GitPanel.tsx

```diff
@@ -198,23 +195,101 @@ export class GitPanel extends React.Component<
   };
 
   render() {
-    let filelist: React.ReactElement;
-    let main: React.ReactElement;
-    let sub: React.ReactElement;
-    let msg: React.ReactElement;
+    return (
+      <div className={panelWrapperClass}>
+        {this._renderToolbar()}
+        {this._renderMain()}
+      </div>
+    );
+  }
 
-    if (this.state.isHistoryVisible) {
-      sub = (
-        <HistorySideBar
-          isExpanded={this.state.isHistoryVisible}
-          branches={this.state.branches}
-          pastCommits={this.state.pastCommits}
-          model={this.props.model}
-          renderMime={this.props.renderMime}
-        />
+  /**
+   * Renders a toolbar.
+   *
+   * @returns fragment
+   */
+  private _renderToolbar = () => {
```

Comment:
Just curious, why do you prefer these as anonymous functions instead of methods?

Reply:
Sorry I guess these are called "arrow functions" not anonymous functions.

Reply:
@saulshanabrook I don't actually prefer using anonymous arrow function expression assignments. I think I was initially following precedent elsewhere in the codebase (when in Rome...), but, upon reflection, I realized that their usage obviates the need for explicit `this` binding for event handlers. I went ahead and changed all methods which automatically have the correct `this` context, but kept the arrow function expressions for those situations where we want the `this` context to refer to the class instance, which happens to exclusively be for DOM event handlers.

I suppose an additional concern for not originally doing this was ensuring that all methods "looked" consistent. One subtlety by making this distinction is that, for those new to the code base, not necessarily obvious why one "style" is chosen over the other, unless readers know how `this` binding works. We could use `function#bind`, but this has been problematic historically due to perf and browser quirks.

Reply:
Thanks again. It might be a good idea to make a JupyterLab wide style for these types of choices, since the distinctions seem rather subtle (hint: I don't get it).

Reply:
@saulshanabrook Re: arrow/anonymous functions. Actually, they were both. :) 

Re: styles. Well, if it were up to me... :) ...I'd prohibit all anonymous functions, all arrow functions, and all function expressions, and only allow function declarations. In other JS projects, we've configured ESLint to enforce such prohibitions. However, in today's JS and especially for Python devs who love lambdas, instituting this sort of change in a project such as JupyterLab is likely a nonstarter.